e2e/nomostest/reset.go

```diff
@@ -39,6 +40,32 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+// sharedTestNamespaces is a list of namespaces that should not be deleted or
+// reset between tests in a shared environment.
+//
+// Reset will skip deletion of these namespaces, if they exist.
+var sharedTestNamespaces = []string{
+	configsync.ControllerNamespace,
+	configmanagement.RGControllerNamespace,
+	metrics.MonitoringNamespace,
+	testGitNamespace,
+}
+
+// protectedNamespaces is a list of namespaces that should never be deleted.
+//
+// Reset will error if these namespaces exist and have the test label.
+//
+// Individual test cleanup MUST revert these namespaces, if modified.
+// See `checkpointProtectedNamespace` for an example.
+var protectedNamespaces = func() []string {
```

Comment:
I didn't mind having these arrays returned by a function as it ensures you always get a pseudo-immutable value (even if it's a bit suboptimal), but I don't feel too strongly about it

Reply:
I suspect it's less important for a private variable where you control all usages. But yeah, it's sort of a language failing that there's no immutable collections you can use as constant.

Java users fixed this problem by adding libraries for ImmutableList and such, but that's hard to do in Go where slices and arrays aren't structs and you can't override their methods or provide alternate impls for their interfaces.